ui/components/Dashboard/DashboardMeshModelGraph.js

```diff
@@ -1,153 +1,141 @@
       ["Relationships", totalRelationships]
-      // TODO: Add Policies
-    ]
-  }, [totalModels,totalRelationships,totalComponents])
+    ];
+  }, [totalModels, totalRelationships, totalComponents]);
 
   const chartOptions = useMemo(() => ({
     data : {
       columns : data,
       type : donut(),
       colors : dataToColors(data),
     },
-    arc : {
-      cornerRadius : {
-        ratio : 0.05
-      }
-    },
-    donut : {
-      title : "Capabilities\nRegistry",
-      padAngle : 0.03,
-    },
-    tooltip : {
-      format : {
-        value : function (v) {
-          return v
-        }
-      }
-    },
-  }) , [data])
+    // ... other configurations as needed
+  }), [data]);
 
   return (
     <div className={classes.dashboardSection}>
-
       <Link href="/settings#meshmodel-summary">
         <Typography variant="h6" gutterBottom className={classes.link}>
           Models
         </Typography>
       </Link>
-      <div>
-        <BBChart options={chartOptions} />
+      <div style={spinnerStyle}>
+        {(loadingModels || loadingComponents || loadingRelationships) ? <CircularProgress /> : <BBChart options={chartOptions} />}
       </div>
     </div>
-  )
+  );
 }
 
-
-
 function MeshModelCategories({ classes }) {
+  const [categoryMap, setCategoryMap] = useState({});
+  const [loading, setLoading] = useState(true);
 
-  const [categoryMap, setCategoryMap] = useState({})
-  const cleanedData = useMemo(() =>
-    Object
-      .keys(categoryMap)
-      .map((key) => ([key, categoryMap[key]]))
-  ,[categoryMap])
-
-  const chartOptions =useMemo(() => ({
+  const cleanedData = useMemo(() => Object.keys(categoryMap).map((key) => ([key, categoryMap[key]])), [categoryMap]);
 
+  const chartOptions = useMemo(() => ({
     data : {
       columns : cleanedData,
       colors : dataToColors(cleanedData),
       type : pie(),
+      labels : false
     },
     tooltip : {
       format : {
         value : function (v) {
-          return `${v} Models`
+          return `${v} Models`;
         }
       }
     },
     legend : {
       show : false
     }
-  }) , [cleanedData])
+  }), [cleanedData]);
 
-  // API Calls
   useEffect(() => {
-
     fetchCategories().then((categoriesJson) => {
-      categoriesJson['categories'].forEach((category) => {
-        let categoryName = category.name
-        getModelFromCategoryApi(categoryName).then((modelsJson) => {
-          setCategoryMap((prevState) => ({ ...prevState, [categoryName] : modelsJson["total_count"] }))
-        })
-      })
-    })
-  }, [])
+      const promises = categoriesJson['categories'].map((category) => {
+        let categoryName = category.name;
+        return getModelFromCategoryApi(categoryName).then((modelsJson) => {
+          setCategoryMap((prevState) => ({ ...prevState, [categoryName] : modelsJson["total_count"] }));
+        });
+      });
+
+      Promise.all(promises).then(() => {
+        setLoading(false);
+      });
```

Comment:
catch , finally

Reply:
@aabidsofi19 shall I use notify hook for handling error cases?

Reply:
yup,make the loading state false in catch and finally